### PR TITLE
Fix service name is null; Throw exception if no service version

### PR DESCRIPTION
**Pull request by qwordy**
Resolve https://github.com/Azure/autorest.java/issues/1185

There are multiple usages of `settings.getServiceName()` in the generator, especially for building service version related components. `--service-name` is an optional argument which could lead to NullPointerException.
It will throw exception if no service version now.
Create an [issue](https://github.com/Azure/autorest.java/issues/1196), Load default service version from swagger file.

### CODE REVIEW

File: javagen/src/main/java/com/azure/autorest/Javagen.java [hunk 1]
```diff
@@ -142,7 +142,12 @@ protected NodeTuple representJavaBeanProperty(Object javaBean, Property property
             // Service version
             if (settings.isLowLevelClient() && settings.getServiceVersions() != null) {
                 String packageName = settings.getPackage();
-                String serviceName = settings.getServiceName().replaceAll("\\s", "");
+                String serviceName;
```
Comment (haolingdong-msft) [inline javagen/src/main/java/com/azure/autorest/Javagen.java:145]: do we need to add a test for the null case?

File: javagen/src/main/java/com/azure/autorest/mapper/ServiceClientMapper.java [hunk 2]
```diff
@@ -125,7 +125,12 @@ public ServiceClient map(CodeModel codeModel) {
             String serviceClientPropertyDefaultValueExpression = serviceClientPropertyClientType.defaultValueExpression(p.getClientDefaultValue());
 
             if (settings.isLowLevelClient() && serviceClientPropertyName.equals("apiVersion")) {
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/mapper/ServiceClientMapper.java:127]: Do we need to test `settings.getServiceVersions() != null`?

File: javagen/src/main/java/com/azure/autorest/Javagen.java (hunk 1 above)
Comment (qwordy) [reply javagen/src/main/java/com/azure/autorest/Javagen.java:145]: Yes, we do.

File: javagen/src/main/java/com/azure/autorest/mapper/ServiceClientMapper.java (hunk 2 above)
Comment (qwordy) [reply javagen/src/main/java/com/azure/autorest/mapper/ServiceClientMapper.java:127]: Yes, we do.

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by haolingdong-msft — APPROVED** [review]
